app/controllers/volunteers_controller.rb

```diff
@@ -62,15 +62,19 @@ def index
     respond_to do |format|
       format.json {
         @volunteers = Volunteer.select('email,id,name,phone').collect do |volunteer|
-          (volunteer.regions.collect{ |region| region.id } &
-            current_volunteer.region_ids).length > 0 ? volunteer : nil
+          volunteer_region_matches = (volunteer.regions.collect{ |region| region.id } & current_volunteer.region_ids).length > 0
+          if volunteer_region_matches
+            volunteer
+          end
```

Comment:
This last if seems like it could be a one liner.

What was the specific complaint from Rubocop on the initial version? 

I wonder if this reads a bit more clearly:

```
volunteer_regions = volunteer.regions.collect{ |region| region.id } # or pluck(:id)? 
if volunteer_regions & current_volunteer.region_ids).length > 0
  volunteer
end
```

What do you think?

Reply:
I think pluck works great here.  Again, all we are trying to do is see if there is an intersection between the volunteer region ids and the current volunteer's region ids.  I wonder  if there's a better way to do this in activerecord.  

```
region_ids = current_volunteer.region_ids
Volunteer.where(region_id: region_ids)
```

Isn't that equivalent?  Because that generates

`where volunteer.region_id in ([region_ids])`, which seems equivalent to me.

Maybe would want to do a few more tests before making that jump.

Reply:
I think this is the same `Volunteer.joins(:regions).where(regions: {id: admin_region_ids})` ?

I get `volunteers.region_id does not exist` when running your where because it is a has many through on assignments: 
`has_many :regions, through: :assignments`.

I still get my head turned around with has many... would `Volunteer.joins(:regions).where(regions: {id: admin_region_ids})` still work?

Reply:
By using region_ids instead of just regions, we are forcing ActiveRecord to run a query and return an Array, and then we use that Array in the next query. I think it may be more efficient to use regions and then let ActiveRecord decide when to make the query, as this would (I think) result in just one query. Also, I think we want to add distinct to the query to avoid duplicate records:

`Volunteer.joins(:regions).where(regions: {id: current_volunteer.regions}).distinct`

This should result in a single query looking something like:

`SELECT DISTINCT "volunteers".* FROM "volunteers" INNER JOIN "assignments" ON "assignments"."volunteer_id" = "volunteers"."id" INNER JOIN "regions" ON "regions"."id" = "assignments"."regions_id" WHERE "regions"."id" IN (SELECT "regions"."id" FROM "regions" INNER JOIN "assignments" ON "regions"."id" = "assignments"."region_id" WHERE "assignments"."volunteer_id" = $1)`

And it should return an ActiveRecord Relation, allowing us to further process before sending. But I may be missing something, as I haven't been close to this project. What do you think?